app/ht_indexer/src/ht_queue_service/queue_connection.py

```diff
@@ -19,50 +24,99 @@ class QueueConnection:
+        When the number reaches the configured count, RabbitMQ will stop delivering more messages on the
+        channel until at least one of the outstanding ones is acknowledged.
+        """
+        # Get a channel
         ht_channel = self.queue_connection.channel()
 
-        # exchange - this can be assumed as a bridge name which needed to be declared so that queues can be accessed
         # declare the exchange
-        # Direct – the exchange forwards the message to a queue based on a routing key
         ht_channel.exchange_declare(self.exchange, durable=True, exchange_type="direct", auto_delete=False)
 
-        ht_channel.queue_bind(self.queue_name, self.exchange, routing_key=self.queue_name)
-
-        # The value defines the max number of unacknowledged deliveries that are permitted on a channel.
-        # When the number reaches the configured count, RabbitMQ will stop delivering more messages on the
-        # channel until at least one of the outstanding ones is acknowledged.
         ht_channel.basic_qos(prefetch_count=self.batch_size)
 
         return ht_channel
 
     def queue_reconnect(self):
-        self.__init__(self.credentials.username, self.credentials.password, self.host, self.queue_name)
 
-    def get_total_messages(self):
-        # durable: Survive reboots of the broker
-        # passive: Only check to see if the queue exists and raise `ChannelClosed` if it doesn't
-        status = self.ht_channel.queue_declare(self.queue_name, durable=True, passive=True)
-        return status.method.message_count
+        """ Reconnect to RabbitMQ server """
+
+        logger.info(f"Reconnecting to RabbitMQ at {self.host}")
+        try:
+            if self.queue_connection and not self.queue_connection.is_closed:
+                self.queue_connection.close()
+        except Exception:
+            pass
+        self._connect()
+
+    def close(self):
+        """ Close the connection to RabbitMQ server """
+        try:
+            if self.queue_connection and not self.queue_connection.is_closed:
+                self.queue_connection.close()
+        except Exception as e:
+            logger.warning(f"Error closing RabbitMQ connection: {e}")
+
+    def is_ready(self) -> bool:
+        return self.queue_connection and self.ht_channel and not self.ht_channel.is_closed
+
+    def get_total_messages(self) -> int:
+        try:
+            # Ensure a channel is open
+            if not self.is_ready():
+                self.queue_reconnect()
+
+            # Use passive=True to avoid creating a queue if it doesn't exist
+            status = self.ht_channel.queue_declare(
+                queue=self.queue_name, durable=True, passive=True
+            )
+            return status.method.message_count
+        # This exception will catch the issue when the queue does not exist or the queue
+        #is declared with different arguments or some permission issue.
+        except pika.exceptions.ChannelClosedByBroker as e:
```

Comment:
Do we need these exception handlers for other things like consuming a message? If so, as a future change it might make sense to extract these into a separate method that's used as a wrapper for calling other operations on the queue.